src/consent-manager-builder/index.js

```diff
@@ -76,8 +78,11 @@ export default class ConsentManagerBuilder extends Component {
   }
 
   componentDidMount() {
-    // TODO: handle errors properly
-    this.initialise()
+    const {onError} = this.props
+    const intialize = this.initialise()
+    if (onError && typeof onError === 'function') {
+      intialize.catch(e => onError(e))
```

Comment:
Is this guaranteed to always be async?
I'm guessing `async/await`ing here might yield safer results.

We could `async/await`, call `await onError(e)` (in case it's also an async function) and then rethrow the error.

cc @mshwery, @Rowno

Reply:
It looks like it's always async, unless people are fudging with the internals. But yeah you could do use `async/await` for additional safety.

Reply:
You mean `initialize`?
That one is always async.

How do you feel about this
```JS
  async componentDidMount() {
    const {onError} = this.props
    if (onError && typeof onError === 'function') {
      try {
        await this.initialise()
      } catch(e) {
        await onError(e)
      }
    } else {
      this.initialise()
    }
  }
```

Reply:
![2019-04-04 14 11 42](https://user-images.githubusercontent.com/952992/55589735-e6220700-56e5-11e9-8125-a75305ae777e.gif)